lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js

```diff
@@ -54,19 +77,43 @@ const runDetectorAsyncQualityDisplay = (runNumber, dplDetectorId, qcSummary) =>
         content,
     );
 
+    let qcSummaryDisplay = null;
     if (!badEffectiveRunCoverage) {
         if (explicitlyNotBadEffectiveRunCoverage === 1) {
-            return getQcSummaryDisplayWithColor(notBadPercentageFormat, { color: ChartColors.Green.dark });
+            qcSummaryDisplay = getQcSummaryDisplayWithColor(notBadPercentageFormat, { color: ChartColors.Green.dark });
         } else {
-            return getQcSummaryDisplayWithColor(notBadPercentageFormat, { color: ChartColors.Gray.dark });
+            qcSummaryDisplay = getQcSummaryDisplayWithColor(notBadPercentageFormat, { color: ChartColors.Gray.dark });
         }
     } else {
         if (badEffectiveRunCoverage === 1) {
-            return getQcSummaryDisplayWithColor(notBadPercentageFormat, { color: ChartColors.Red.dark });
+            qcSummaryDisplay = getQcSummaryDisplayWithColor(notBadPercentageFormat, { color: ChartColors.Red.dark });
         } else {
-            return getQcSummaryDisplayWithColor(notBadPercentageFormat, { color: ChartColors.Orange.dark });
+            qcSummaryDisplay = getQcSummaryDisplayWithColor(notBadPercentageFormat, { color: ChartColors.Orange.dark });
         }
     }
+
+    if (dataPassId) {
+        return frontLink(
```

Comment:
This can also be extracted in a similar fashion to the comment above

Reply:
done